### ISSUE TITLE
Unexpected Transitive Tagging

### ISSUE THREAD

**Issue by hannesfrank** [remnoteio/remnote-issues#199]
**Describe the bug**
When tags are applied like shown (A #B, B #C) then A is also #C. Is this intended behavior?

It might be useful when using tags as subclasses, e.g. tag a book genre `fantasy` with `#book` and a book title with `#fantasy` and you have all books collected in `book`. But it confused me and maybe this could be modeled better and more flexibly with an explicit search query.

**To Reproduce**

![image](https://user-images.githubusercontent.com/4137110/96349567-c4625800-10b0-11eb-92e6-cf10c9cde5a1.png)
These are the references of C, and A does only have `#B` and not `#C`. Some waiting and/or reload might be required. I could not make the MWE work the first time I tried but it shows the behavior now.

**Comment by ognsya** [comment]
I agree this is a very useful feature, but one might not want/expect it.

**Comment by hannesfrank** [comment]
I would assume this is because the inheritance semantic of tags - the same thing that powers templates i.e. having `Book #Publication` and `How to Take Smart Notes #Book` lists the latter as `#Publication` as well.

There is an inconsistency related to [Rem Level Formatting](https://forum.remnote.io/t/custom-rem-level-formatting-with-tags/168?u=hannesfrank): The `rem-data-tags` are not applied transitively as well, i.e. `How to Take Smart Notes` does not have the `rem-data-tag` aka _Type_ of `Publication`.

**Comment by hannesfrank** [comment]
After playing around a bit with this I find it actually quite cool that we have a type system.

The main point that confused me is the backlink counter (tag counter) which shows not only how many rems are tagged with this rem but also _transitively_ all rems tagged with tagged rem.
I would only show directly tags there and have the transitive version shown in an (extra) search portal stub.

**Comment by hannesfrank** [comment]
The tag counters are inconsistent: The inline counter includes inherited tags (here it shows 3). The tag search portal does not (showing only 1):

![Bildschirmfoto vom 2021-01-22 16-57-50](https://user-images.githubusercontent.com/4137110/105514356-af125080-5cd3-11eb-8811-a656852cfe71.png)

**Comment by metamas** [comment]
@hannesfrank Thank you for digging into this and sharing your findings. The transitive tagging behavior is definitely not what I've been intending as I tag things. I use tags on a variety of one-off rems (e.g. tasks, links, quick thoughts) that are intended to represent a relevance to those tag rems (e.g. concepts, projects, people), not inheritance from them.

I agree that the transitive behavior can be useful, but I would prefer it to be an option next to tagging for relevance. And, yes, further distinction in the search portals would be nice. The transitivity of tags has also confused me in how ancestor slots are listed for rems with descendent tags.

Honsetly, now I'm poking around and feeling unsure about a lot of my established mental model for how tag/reference/slot related features work. Will have to experiment further and probably overhaul hundreds of tags.
